api/api.proto

```diff
@@ -3229,6 +3238,73 @@ message SdkNodeInspectRequest {
   string node_id = 1;
 }
 
+// Options for draining volume attachment from a node
+message SdkNodeDrainAttachmentOptions {
+    // Selector is used for selecting volumes whose
+    // attachment needs to be moved from this node.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+    // Drain only sharedv4 volumes from a node
+    // By default all volumes will be drained.
+    bool only_sharedv4 = 2;
+}
+
+// Options for draining volume replicas from a node
+message SdkNodeDrainVolumeReplicaOptions {
+    // Selector is used for selecting volumes whose
+    // replicas need to be moved from this node to another.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+}
+
+// Options for draining/detaching cloud drives from a node
+message SdkNodeDrainCloudDriveOptions {
+    // When set the cloud drives will also get deleted.
+    bool delete_drives = 1;
```

Comment:
the delete drive part should be part of the node decommission API. So we can keep it out of here.

Reply:
makes sense. done